design/one-pager-change-logs.md

```diff
@@ -0,0 +1,410 @@
+pod logs, and they can also be exported to external/centralized systems for long
+term storage and further inspection and analysis. One suggested means to collect
+and export the change logs is to use the [OpenTelemetry
+Collector](https://opentelemetry.io/docs/collector/), but a more detailed
+analysis of that setup is out of scope for this document.
+
+#### Pod Log Benefits
+
+Pod logs are a good option for the change log location because they:
+
+* are very standardized, with many tools capable of reading, interacting with,
+  and scraping them
+* have built in management features like log rotation
+* provide sufficient durability for our needs, e.g. they can survive network
+  outages and pod crashes
+
+### Flow of Change Log Entries
+
+As mentioned in the previous section, change log entries will be written to the
+pod logs. However, there is some additional structure we will build in order to
+keep change logs entirely separated from the regular provider log entries, which
+will aid not only with human readability, but also post processing and analysis
+by the system responsible for long term storage.
+
+Each provider pod that generates change log entries will have a sidecar
+container solely responsible for receiving the entries and writing them to
+`stdout` (i.e. the pod logs for the sidecar container). This sidecar container
+will be listening on a gRPC connection from the main provider container. This
+gRPC connection between the two containers will be established over a Unix
+domain socket on a shared `emptyDir` volume mounted by both containers. This
+setup allows the main provider container to send the change log entries to the
+sidecar container in a fairly performant manner without needing any network
+access at all.
+
+The change log entry data will be serialized in a binary protobuf format for
+transmission over the gRPC connection, while the data will be persisted to the
+pod logs in a JSON serialized format.
+
+#### Summary: Change Log Entry Flow
+
+* The provider will have a sidecar container where each entry is sent
+* Main container will communicate through gRPC with the sidecar container over a
+  Unix domain socket on a shared volume
+* Data is serialized in a protobuf binary format and written to the sidecar
+  container logs in JSON format
+
+### General Architecture
+
+The general architecture of the change log system is captured in the diagram
+below:
+
+![Change logs architecture diagram](./images/change-logs-architecture.png)
+
+### Setup and Configuration
+
+This feature will be declared at an
+[alpha](https://docs.crossplane.io/latest/learn/feature-lifecycle/#alpha-features)
+level of maturity when it is first released, meaning that Crossplane users must
+opt-in to using this functionality by setting an `--enable-change-logs` feature
+flag. Note this flag will need to be set on each provider for which change
+logs are desired, likely via a `DeploymentRuntimeConfig`.
+
```

Comment:
Has there been any discussion of whether this feature will be enabled-by-default after it leaves alpha?  I can see where this will be very useful to a lot of people, but it also seems to (potentially) add a non-trivial amount of overhead that is likely to be unneeded for many use cases.  As long as the extra `Observe()` call is required I think we need to be able to turn off the feature, or require it to be specifically enabled even after alpha.

Reply:
Yes, this will definitely be an alpha feature that folks have to opt-in to enable it!  See https://github.com/jbw976/crossplane/blob/design-change-logs/design/one-pager-change-logs.md#setup-and-configuration